Unify CI workflows

michalinacienciala opened the pull request:
We have CI workflows scattered across various repositories. We try to keep their implementation as similar as possible. In this PR and the ones listed below we introduce a couple of changes which unify used solutions and fix small mistakes, such as:
* making tagging of published packages consistent
* passing `NPM_TOKEN` secret as variable instead of passing it by writing to file
* handling caching of npm/yarn dependencies via `setup-node` action
* adding `workflow_dispatch` job trigger where it's missing
* using Node.js in version 14.x instead of 12.x where possible

Related PRs in other repositories:
* https://github.com/keep-network/keep-core/pull/2776
* https://github.com/keep-network/keep-ecdsa/pull/941
* https://github.com/keep-network/tbtc/pull/842
* https://github.com/keep-network/tbtc.js/pull/145
* https://github.com/keep-network/local-setup/pull/124
* https://github.com/keep-network/coverage-pools/pull/204
* https://github.com/keep-network/tbtc-v2/pull/88
* https://github.com/threshold-network/solidity-contracts/pull/59

nkuba commented on .github/workflows/lint.yml:
```diff
@@ -1,6 +1,6 @@
 name: Run linters
 
-on: [push]
+on: [push, workflow_dispatch]
```
Shouldn't this be as below for consistency with other projects?
```
on:
  push:
    branches:
      - main
  pull_request:
```

michalinacienciala replied:
Yeah, you're right, changed in d451cc4.

michalinacienciala commented:
@nkuba, ready for another review. Apart of addressing your comment in d451cc4ba2dc2b8a1ffc2dfcaf1193727b95a8de I also added three other commits with small changes.
